From 1ad5aa4b69188d8af320193e125c8e099179f9cf Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Timi=20Rautam=C3=A4ki?= <timi.rautamaki@gmail.com>
Date: Wed, 4 May 2022 12:31:56 +0000
Subject: [PATCH] PermissionManager: fix NPE in getIndicatorExemptedPackages

sLocationProviderPkgNames and sLocationExtraPkgNames can be null
because they're static. Populate them if they are null.
Also rename them from member to static variable to follow code style.

Change-Id: I7d08e02047ccf183cc75ce18b3ebd2086b2831b3
---
 .../android/permission/PermissionManager.java | 20 +++++++++++++------
 1 file changed, 14 insertions(+), 6 deletions(-)

diff --git a/core/java/android/permission/PermissionManager.java b/core/java/android/permission/PermissionManager.java
index 4ccadc89c99f..fe3f72d2c260 100644
--- a/core/java/android/permission/PermissionManager.java
+++ b/core/java/android/permission/PermissionManager.java
@@ -145,8 +145,8 @@ public final class PermissionManager {
 
     private List<SplitPermissionInfo> mSplitPermissionInfos;
 
-    private static String[] mLocationProviderPkgNames;
-    private static String[] mLocationExtraPkgNames;
+    private static String[] sLocationProviderPkgNames;
+    private static String[] sLocationExtraPkgNames;
 
     /**
      * Creates a new instance.
@@ -163,9 +163,9 @@ public final class PermissionManager {
                 "permissionmgr"));
         mLegacyPermissionManager = context.getSystemService(LegacyPermissionManager.class);
 
-        mLocationProviderPkgNames = context.getResources().getStringArray(
+        sLocationProviderPkgNames = context.getResources().getStringArray(
                 R.array.config_locationProviderPackageNames);
-        mLocationExtraPkgNames = context.getResources().getStringArray(
+        sLocationExtraPkgNames = context.getResources().getStringArray(
                 R.array.config_locationExtraPackageNames);
     }
 
@@ -954,12 +954,20 @@ public final class PermissionManager {
                 pkgNames.add(exemptedPackage);
             }
         }
-        for (String pkgName: mLocationProviderPkgNames) {
+        if (sLocationProviderPkgNames == null) {
+            sLocationProviderPkgNames = context.getResources().getStringArray(
+                    R.array.config_locationProviderPackageNames);
+        }
+        for (String pkgName: sLocationProviderPkgNames) {
             if (pkgName != null) {
                 pkgNames.add(pkgName);
             }
         }
-        for (String pkgName: mLocationExtraPkgNames) {
+        if (sLocationExtraPkgNames == null) {
+            sLocationExtraPkgNames = context.getResources().getStringArray(
+                    R.array.config_locationExtraPackageNames);
+        }
+        for (String pkgName: sLocationExtraPkgNames) {
             if (pkgName != null) {
                 pkgNames.add(pkgName);
             }
-- 
2.34.1

